content/blog/2019-09-23-grpc-on-dotnetcore.md

```diff
@@ -0,0 +1,50 @@
+
+## Getting Started
+
+gRPC templates are included as part of the .NET SDK. To get started, navigate to a console window after installing the SDK and run the following commands.
+
+```sh
+dotnet new grpc -o GrpcGreeter
+cd GrpcGreeter
+dotnet run
+```
+
+To create a gRPC client and test with the newly created gRPC Greeter service, you can [follow the rest of this tutorial here](https://docs.microsoft.com/en-us/aspnet/core/tutorials/grpc/grpc-start?view=aspnetcore-3.0&tabs=visual-studio-code).
+
+##  What's new?
+
+Unlike the existing C-Core based implementation ([Grpc.Core](https://github.com/grpc/grpc/tree/master/src/csharp)), the new libraries ([grpc-dotnet](https://github.com/grpc/grpc-dotnet)) make use of the existing networking primitives in the .NET Core Base Class Libraries (BCL). The diagram below highlights the difference between the existing **Grpc.Core** library and the new **grpc-dotnet** libraries.
+
+<p><img src="/img/grpc-dotnet.svg" alt="gRPC .NET Stack" style="max-width: 800px" /></p>
+
+On the server side, the `Grpc.AspNetCore.Server` package integrates into ASP.NET Core, allowing developers to benefit from ecosystem of common cross-cutting concerns of logging, configuration, dependency injection, authentication, authorization etc. which have already been solved by ASP.NET Core. Popular libraries in the ASP.NET ecosystem such as [Entity Framework Core (ORM)](https://github.com/aspnet/EntityFrameworkCore), [Serilog (Logging library)](https://github.com/serilog/serilog), and [Identity Server](https://github.com/IdentityServer/IdentityServer4) among others now work seamlessly with gRPC.
+
+On the client side, the `Grpc.Net.Client` package builds upon the familiar `HttpClient` API that ships as part of .NET Core. As with the server, gRPC clients greatly benefit from the ecosystem of packages that build upon `HttpClient`. It is now possible to use existing packages such as [**Polly**(Resilience and fault-handling library)](https://github.com/App-vNext/Polly) and [HttpClientFactory(manage HTTPClient lifetimes)](https://github.com/aspnet/Extensions/tree/master/src/HttpClientFactory) with gRPC clients.
+
+The diagram below captures the exhaustive list of all new .NET packages for gRPC and their relationship with the existing packages.
+
+<p><img src="/img/grpc-dotnet-packages.svg" alt="grpc-dotnet packages" style="max-width: 800px" /></p>
+
+In addition to the new published packages that ship as part of **grpc-dotnet**, we've also made improvements that benefit both stacks. There have been improvements to the existing packages including a more cohesive experience when working in Visual Studio especially around design-time build.
```

Comment:
Call out syntax highlighting in VS and VS Code? Can add a screenshot of it in action. It's nice to show off a feature that people can just *see*.

Reply:
It's only VS, not VS Code. Still worth mentioning?

Reply:
Up to you! I take any change I can get to add pictures to blog posts to make them visually engaging :D

Reply:
Done!